File: Section 1 Introduction/maximum_pairwise_product_naive.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter numbers separated by spaces: ", function(input) {
    const numbers = input.split(" ").map(Number);

    if (numbers.length < 2) {
        console.log("Please enter at least two numbers.");
      } else {
        let max = -Infinity;
    
        for (let i = 0; i < numbers.length; i++) {
          for(let j = i + 1; j < numbers.length; j++) {
            const product = numbers[i] * numbers[j];
            if (product > max) {
                max = product;
            }
          }
        }
    
        console.log("Maximum pairwise product is:", max);
      }
    
      rl.close();
})

File: Section 1 Introduction/programming-challenges/maximum_pairwise_product_naive.py
def maximum_pairwise_product(numbers):
    max_product = 0
    for i in range(len(numbers)):
        for j in range(i + 1, len(numbers)):
            max_product = max(max_product, numbers[i] * numbers[j])

    return max_product


n = int(input())
x = input().split()
a = []
for i in range(len(x)):
    a.append(int(x[i]))

print(maximum_pairwise_product(a))

"""
def maximum_pairwise_product(numbers):
    max_product = 0
    for i in range(len(numbers)):
        for j in range(i + 1, len(numbers)):
            max_product = max(max_product, numbers[i] * numbers[j])
    return max_product

n = int(input("Enter number of elements: "))
x = list(map(int, input("Enter the numbers separated by space: ").split()))

if len(x) != n:
    print("Error: Number of elements doesn't match.")
else:
    print("Maximum pairwise product is:", maximum_pairwise_product(x))
"""